src/basic/copy.h

```diff
@@ -57,6 +57,9 @@ static inline int copy_directory(const char *from, const char *to, CopyFlags cop
         return copy_directory_full(from, to, copy_flags, NULL, NULL, NULL);
 }
 
+int acquire_data_fd(const void *data, size_t size, unsigned flags);
+int copy_data_fd(int fd);
```

Comment:
hmm, do they really fit into copy.c? 

maybe just give them their own .c/.h file? they are complex enough for that. i.e. `data-fd-util.[ch]`?

Reply:
OK, will do.